tests/security/selinux/selinux_setup.pm

```diff
@@ -13,109 +12,63 @@ use testapi;
         'setools-devel', 'setools-java', 'setools-libs', 'setools-tcl',
-        'policycoreutils-python-utils'
+        'policycoreutils-python-utils', 'policycoreutils-python', 'libselinux1-32bit', 'selinux-policy-minimum',
+        'selinux-policy', 'selinux-policy-minimum', 'selinux-policy-devel'
     );
-    if (!is_alp) {
-        foreach my $pkg (@pkgs) {
-            my $results = script_run("zypper --non-interactive se $pkg");
-            if ($results) {
-                record_info('WARNING', "Package $pkg is missing, zypper search returns: $results");
-            }
-            else {
-                zypper_call("in $pkg");
-            }
-        }
-        if (is_x86_64) {
-            zypper_call('in libselinux1-32bit');
-        }
-    }
-    # For opensuse, e.g, Tumbleweed install selinux_policy pkgs as needed
+    zypper_install_available "@pkgs";
+
     # For sle15 and sle15+ "selinux-policy-*" pkgs will not be released
-    # NOTE: have to install "selinux-policy-minimum-*" pkg due to this bug: bsc#1108949
-    # Install policy packages separately due to this bug: bsc#1177675
-    if (is_sle('>=15-sp4')) {
-        # Download and install selinux policy packages
-        assert_script_run("wget --quiet --no-check-certificate $repo_link/$policy_pkg_20220124");
-        assert_script_run("tar -xzf $policy_pkg_20220124");
-        assert_script_run("rpm -ivhU --nosignature --nodeps --noplugins ./selinux*20220124*.rpm", timeout => 120);
-    }
-    elsif (is_sle('>=15') && is_sle('<15-sp4')) {
-        # Download and install selinux policy packages
-        assert_script_run("wget --quiet --no-check-certificate $repo_link/$policy_pkg_20200219");
-        assert_script_run("tar -xzf $policy_pkg_20200219");
-        assert_script_run("rpm -ivhU --nosignature --nodeps --noplugins ./selinux*20200219*.rpm", timeout => 120);
-    }
-    elsif (!is_sle && !is_leap && !is_alp) {
-        my @files = ('selinux-policy', 'selinux-policy-minimum', 'selinux-policy-devel');
-        foreach my $file (@files) {
-            zypper_call("in $file");
-        }
-    }
-    elsif (!is_alp) {
-        zypper_call('in selinux-policy-minimum');
+    # NOTE 1: we have to install "selinux-policy-minimum-*" pkg due to this bug: bsc#1108949
+    # NOTE 2: we have to install policy packages separately due to this bug: bsc#1177675
+    if (is_sle('>=15')) {
+        my $policy_date = get_policy_date();
+        my $policy_archive = $policy_date ? "selinux-$policy_date.tgz" : 'none';
+        assert_script_run("wget --quiet --no-check-certificate https://gitlab.suse.de/qe-security/testing/-/raw/main/data/selinux/$policy_archive");
+        assert_script_run("tar -xzf $policy_archive");
+        assert_script_run("rpm -ivhU --nosignature --nodeps --noplugins ./selinux*$policy_date*.rpm", timeout => 120);
     }
+}
+
+sub run {
+    my ($self) = @_;
+
+    # on ALP selinux is enabled and set to enforcing by default
+    if (is_alp || is_sle_micro('>=6.0')) {
```

Comment:
`is_sle_micro('>=6.0'))` will be alp so I don't think it is needed.

Reply:
will it be ALP or will it be SLE Micro with alp codebase? 🤔  I honestly have no idea, but we can clean it up once we know more